File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/resources/FontResourceFnt.java
package pize.tests.minecraft.game.resources;

import pize.graphics.font.BitmapFont;
import pize.graphics.font.FontLoader;
import pize.tests.minecraft.utils.log.Logger;

public class FontResourceFnt extends Resource<BitmapFont>{

    private BitmapFont font;
    private boolean loaded;

    public FontResourceFnt(String location){
        super(location);
    }

    @Override
    public void loadResource(){
        try{
            font = FontLoader.loadFnt(getLocation());
            loaded = true;
        }catch(RuntimeException e){
            loaded = false;
            Logger.instance().warn(e.getLocalizedMessage());
        }
    }

    @Override
    public void reloadResource(){
        BitmapFont oldFont = font;
        loadResource();
        oldFont.dispose();
    }

    @Override
    public boolean isLoaded(){
        return loaded;
    }

    @Override
    public BitmapFont getResource(){
        return font;
    }

    @Override
    public void dispose(){
        font.dispose();
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/resources/FontResourceFnt.kt
package pize.tests.minecraft.game.resources

import pize.graphics.font.BitmapFont
import pize.graphics.font.FontLoader.loadFnt
import pize.tests.minecraft.utils.log.Logger

class FontResourceFnt(location: String) : Resource<BitmapFont?>(location) {
    override var resource: BitmapFont? = null
        private set
    override var isLoaded = false
        private set

    override fun loadResource() {
        try {
            resource = loadFnt(location)
            isLoaded = true
        } catch (e: RuntimeException) {
            isLoaded = false
            Logger.Companion.instance().warn(e.localizedMessage)
        }
    }

    override fun reloadResource() {
        val oldFont = resource
        loadResource()
        oldFont!!.dispose()
    }

    override fun dispose() {
        resource!!.dispose()
    }
}